cmd/vm/switch_linux.go

```diff
@@ -0,0 +1,230 @@
+		}
+	}()
+
+	return <-errCh
+}
+
+func loopbackUp() error {
+	lo, err := netlink.LinkByName("lo")
+	if err != nil {
+		return err
+	}
+
+	return netlink.LinkSetUp(lo)
+}
+
+func linkUp(iface, mac string) error {
+	link, err := netlink.LinkByName(iface)
+	if err != nil {
+		return err
+	}
+	if mac == "" {
+		return netlink.LinkSetUp(link)
+	}
+	hw, err := net.ParseMAC(mac)
+	if err != nil {
+		return err
+	}
+	if err := netlink.LinkSetHardwareAddr(link, hw); err != nil {
+		return err
+	}
+
+	logrus.Debugf("successful link setup %+v\n", link)
+	return netlink.LinkSetUp(link)
+}
+
+func dhcp(iface string) error {
+	if _, err := exec.LookPath("udhcpc"); err == nil { // busybox dhcp client
+		cmd := exec.Command("udhcpc", "-f", "-q", "-i", iface, "-v")
+		cmd.Stderr = os.Stderr
+		cmd.Stdout = os.Stdout
+		return cmd.Run()
+	}
+	cmd := exec.Command("dhclient", "-4", "-d", "-v", iface)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	return cmd.Run()
+}
+
+func rx(conn io.Writer, tap *water.Interface, errCh chan error, mtu int) {
+	logrus.Info("waiting for packets...")
+	var frame ethernet.Frame
+	for {
+		frame.Resize(mtu)
+		n, err := tap.Read([]byte(frame))
+		if err != nil {
+			errCh <- fmt.Errorf("reading packet from tap failed: %w", err)
+			return
+		}
+		frame = frame[:n]
+
+		size := make([]byte, 2)
+		binary.LittleEndian.PutUint16(size, uint16(n))
+
+		if _, err := conn.Write(size); err != nil {
+			errCh <- fmt.Errorf("writing size to the socket failed: %w", err)
+			return
+		}
+		if _, err := conn.Write(frame); err != nil {
+			errCh <- fmt.Errorf("writing packet to the socket failed: %w", err)
+			return
+		}
+
+		if debug {
+			packet := gopacket.NewPacket(frame, layers.LayerTypeEthernet, gopacket.Default)
+			logrus.Infof("wrote packet (vm -> host %d): %s", size, packet.String())
+		}
+	}
+}
+
+func tx(conn io.Reader, tap *water.Interface, errCh chan error, mtu int) {
+	sizeBuf := make([]byte, 2)
+	buf := make([]byte, mtu+header.EthernetMinimumSize)
+
+	for {
+		n, err := io.ReadFull(conn, sizeBuf)
+		if err != nil {
+			errCh <- fmt.Errorf("reading size from socket failed: %w", err)
+			return
+		}
+		if n != 2 {
+			errCh <- fmt.Errorf("unexpected size %d", n)
+			return
+		}
+		size := int(binary.LittleEndian.Uint16(sizeBuf[0:2]))
+
+		if cap(buf) < size {
+			buf = make([]byte, size)
+		}
+
+		n, err = io.ReadFull(conn, buf[:size])
+		if err != nil {
+			errCh <- fmt.Errorf("reading payload from socket failed: %w", err)
+			return
+		}
+		if n == 0 || n != size {
+			errCh <- fmt.Errorf("unexpected size %d != %d", n, size)
+			return
+		}
+
+		if _, err := tap.Write(buf[:size]); err != nil {
+			errCh <- fmt.Errorf("writing packet to tap failed: %w", err)
+			return
+		}
+
+		if debug {
+			packet := gopacket.NewPacket(buf[:size], layers.LayerTypeEthernet, gopacket.Default)
+			logrus.Infof("read packet (host -> vm %d): %s", size, packet.String())
+		}
+	}
+}
+
+func checkForExsitingIf(ifName string) error {
```

Comment:
`If` sounds like the English conjunction. Can we be slightly more verbose here and use `iface` or `interface`?